### PR TITLE
Add maturity model

### PR DESCRIPTION
**Pull request by roryqi**

### CODE REVIEW

File: community/maturity.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+### CODE
+
+| **ID**   | **Description**                                                                                                                                                                                                                                                | **Status**                                                                                                                                                                                                          |
+| -------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (LuciferYang) [inline community/maturity.md]: use IntelliJ to reformat the table?

File: community/maturity.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+| **CD10** | The project produces Open Source software for distribution to the public, at no charge.                                                                                                                                                                        | **YES** The project source code is licensed under the `Apache License 2.0`.                                                                                                                                         |
```
Comment (LuciferYang) [inline community/maturity.md]: `**YES** The project's source code`

File: community/maturity.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+| **CD20** | Anyone can easily discover and access the project's code..                                                                                                                                                                                                     | **YES** The [official website](https://uniffle.apache.org/) includes `GitHub` link which can access the project's repository on GitHub directly.                                                                    |
+| **CD30** | Anyone using standard, widely-available tools, can build the code in a reproducible way.                                                                                                                                                                       | **YES**  Apache OpenDAL provide `how-to-build` document for every component to tell user how to compile on bare metal, such as the [core's](https://github.com/apache/opendal/blob/main/core/CONTRIBUTING.md).      |
```
Comment (LuciferYang) [inline community/maturity.md]: Seems there's an extra space between `**YES**` and `Apache OpenDAL`? 
And why is it `Apache OpenDAL` here? There are about 10 instances where it's written as `OpenDAL` in this file. Is it a typo?

File: community/maturity.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+| **LC30** | The libraries mentioned in LC20 are available as Open Source software.                                                                                                            | **YES** All dependencies are listed are available as Open Source software                                                                                                   |
+| **LC40** | Committers are bound by an Individual Contributor Agreement (the "Apache iCLA") that defines which code they may commit and how they need to identify code that is not their own. | **YES** All committers have iCLAs.                                                                                                                                          |
+| **LC50** | The project clearly defines and documents the copyright ownership of everything that the project produces.                                                                        | **YES** And all source files are with APLv2 header, checked by `rat`.                                                                                                         |
```
Comment (LuciferYang) [inline community/maturity.md]: Is the `And` here redundant?

File: community/maturity.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+| -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **RE10** | Releases consist of source code, distributed using standard and open archive formats that are expected to stay readable in the long term.                              | **YES** Source release is distributed via [dist.apache.org](https://dist.apache.org/repos/dist/release/uniffle/) and linked from [download page](https://uniffle.apache.org/download/).  |
+| **RE20** | The project's PPMC (Project Management Committee, see CS10) approves each software release in order to make the release an act of the Foundation.                      | **YES** All releases have been voted at dev@opendal.apache.org and general@incubator.apache.org, and have at least 3 PPMC member's votes.                                                |
```
Comment (LuciferYang) [inline community/maturity.md]: `dev@opendal.apache.org`?

Comment (roryqi) [reply community/maturity.md]: I will correct this. This is a draft now.

File: community/maturity.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+| **LC10** | The Apache License, version 2.0, covers the released code.                                                                                                                        | **YES** The [LICENSE](https://github.com/apache/incubator-uniffle/blob/master/LICENSE) is in GitHub repository. And all source files are with APLv2 header, checked by `rat`. |
+| **LC20** | Libraries that are mandatory dependencies of the project's code do not create more restrictions than the Apache License does.                                                     | **YES** All dependencies are listed.                                                                                                                                          |
+| **LC30** | The libraries mentioned in LC20 are available as Open Source software.                                                                                                            | **YES** All dependencies are listed are available as Open Source software                                                                                                     |
```
Comment (advancedxy) [inline community/maturity.md]: nit: --> `All dependencies listed are`

File: community/maturity.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+| **LC40** | Committers are bound by an Individual Contributor Agreement (the "Apache iCLA") that defines which code they may commit and how they need to identify code that is not their own. | **YES** All committers have iCLAs.                                                                                                                                            |
+| **LC50** | The project clearly defines and documents the copyright ownership of everything that the project produces.                                                                        | **YES** And all source files are with APLv2 header, checked by `rat`.                                                                                                         |
```
Comment (advancedxy) [inline community/maturity.md]: Is this the right description for LC50 status?
I think it asks for confirmation of everything this project produces, which should include binary releases? Simply stating all source files are with APLv2 header seems that it cannot answer that.

File: community/maturity.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+| **CS50** | All "important" discussions happen asynchronously in written form on the project's main communications channel. Offline, face-to-face or private discussions that affect the project are also documented on that channel. | **YES** All important discussions and conclusions are recorded in written form.                            |
+
+### Independence
+
+| **ID**   | **Description**                                                                          | **Status**                                                                                                                                                              |
+| -------- | ---------------------------------------------------------------------------------------- |-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **IN10** | The project is independent from any corporate or organizational influence.               | **YES** The PPMC members and committer of Apache Uniffle are from several different companies, and majority of them are NOT From the company that donated this project. |
```
Comment (advancedxy) [inline community/maturity.md]: `Not From`?
---> `not from`?

File: community/maturity.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+| **CD10** | The project produces Open Source software for distribution to the public, at no charge.                                                                                                                                                                        | **YES** The project source code is licensed under the `Apache License 2.0`.                                                                                                                                                 |
+| **CD20** | Anyone can easily discover and access the project's code..                                                                                                                                                                                                     | **YES** The [official website](https://uniffle.apache.org/) includes `GitHub` link which can access the project's repository on GitHub directly.                                                                            |
```
Comment (rickyma) [inline community/maturity.md]: Remove a `.`:
Anyone can easily discover and access the project's code..
->
Anyone can easily discover and access the project's code.

File: community/maturity.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+| **CD30** | Anyone using standard, widely-available tools, can build the code in a reproducible way.                                                                                                                                                                       | **YES**  Apache Uniffle provides `how-to-build` document for every component to tell user how to compile on bare metal, such as the [core's](https://github.com/apache/incubator-uniffle/blob/master/core/CONTRIBUTING.md). |
```
Comment (rickyma) [inline community/maturity.md]: Apache Uniffle provides a \`how-to-build\` document for each component to guide users on how to compile on bare metal, such as the \[core's document\](https://github.com/apache/incubator-uniffle/blob/master/core/CONTRIBUTING.md).

File: community/maturity.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+### Quality
+
+| **ID**   | **Description**                                                                                                                                                                               | **Status**                                                                                                                                                                             |
+| -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **QU10** | The project is open and honest about the quality of its code. Various levels of quality and maturity for various modules are natural and acceptable as long as they are clearly communicated. | **YES** We encourage user to [report issues](https://github.com/apache/incubator-uniffle/issues).                                                                                      |
```
Comment (rickyma) [inline community/maturity.md]: We encourage user to
->
We encourage users to

File: community/maturity.md [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+| -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- |---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **CO10** | The project has a well-known homepage that points to all the information required to operate according to this maturity model.                                         | **YES** The [official website](https://uniffle.apache.org/) includes all information user need to run Apache Uniffle.                                                                                                     |
```
Comment (rickyma) [inline community/maturity.md]: includes all information user need to run
->
includes all information users need to run

File: community/maturity.md (hunk 8 above)
Comment (rickyma) [reply community/maturity.md]: The PPMC members and committers of Apache Uniffle are from several different companies, and the majority of them are NOT from the company that donated this project.

File: community/maturity.md [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+| **RE20** | The project's PPMC (Project Management Committee, see CS10) approves each software release in order to make the release an act of the Foundation.                      | **YES** All releases have been voted at dev@uniffle.apache.org and general@incubator.apache.org, and have at least 3 PPMC member's votes.                                                |
+| **RE30** | Releases are signed and/or distributed along with digests that anyone can reliably use to validate the downloaded archives.                                            | **YES** All releases are signed, and the [KEYS](https://downloads.apache.org/incubator/uniffle/KEYS) are available.                                                                      |
+| **RE40** | The project can distribute convenience binaries alongside source code, but they are not Apache Releases, they are provided with no guarantee.                          | **YES**                                                                                                                                                                                  |
```
Comment (maobaolong) [inline community/maturity.md]: Releases -> releases ?

File: community/project committers.md [hunk 14]
```diff
@@ -43,3 +43,19 @@
   | Weiwei Yang     | Apple        |
   | Zhankun Tang    | Cloudera     |
   | Mao Baolong     | Tencent      |
+
+## how to become a committer
```
Comment (maobaolong) [inline community/project committers.md:47]: Beyond describing existing contributors and PPMC members, add a section on how new contributors are attracted, for example:

"To attract new contributors, we regularly host open source contributor meeting, offer a mentorship program, and actively respond to every message on PRs to foster an environment that welcomes and supports newcomers."

File: community/maturity.md (hunk 8 above)
Comment (maobaolong) [reply community/maturity.md]: Provide specific examples of how the project maintains its independence, such as:

"Although the project originated from XYZ Company, the current PPMC members come from X different companies, ensuring the independence and diversity of project decisions."

File: community/maturity.md [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+
+### Consensus
+
+| **ID**   | **Description**                                                                                                                                                                                                           | **Status**                                                                                                 |
+| -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |------------------------------------------------------------------------------------------------------------|
+| **CS10** | The project maintains a public list of its contributors who have decision power. The project's PPMC (Project Management Committee) consists of those contributors.                                                        | **Yes** See [members](https://uniffle.apache.org/community/project%20committers/) with committers.         |
```
Comment (maobaolong) [inline community/maturity.md]: As terminology as been defined at the first mention, it is not necessary to explain it here.

**Review by maobaolong — COMMENTED** [review]
@jerqi Thanks for your effort for this, left some minor comments.

File: community/maturity.md (hunk 2 above)
Comment (roryqi) [reply community/maturity.md]: Correct.

File: community/maturity.md (hunk 3 above)
Comment (roryqi) [reply community/maturity.md]: Correct. Removed OpenDAL.

File: community/maturity.md (hunk 6 above)
Comment (roryqi) [reply community/maturity.md]: Correct.

File: community/maturity.md (hunk 7 above)
Comment (roryqi) [reply community/maturity.md]: I refer to OpenDAL and Celeborn. 
https://cwiki.apache.org/confluence/display/CELEBORN/Apache+Maturity+Model+Assessment+for+Celeborn

https://opendal.apache.org/community/maturity/

File: community/maturity.md (hunk 9 above)
Comment (roryqi) [reply community/maturity.md]: Thanks. Removed.

File: community/maturity.md (hunk 10 above)
Comment (roryqi) [reply community/maturity.md]: OK.

File: community/maturity.md (hunk 11 above)
Comment (roryqi) [reply community/maturity.md]: OK.

File: community/maturity.md (hunk 12 above)
Comment (roryqi) [reply community/maturity.md]: OK.

File: community/maturity.md (hunk 13 above)
Comment (roryqi) [reply community/maturity.md]: OK.

File: community/maturity.md (hunk 4 above)
Comment (roryqi) [reply community/maturity.md]: Removed.

File: community/project committers.md (hunk 14 above)
Comment (roryqi) [reply community/project committers.md:47]: OK.

File: community/maturity.md (hunk 15 above)
Comment (roryqi) [reply community/maturity.md]: We should answer every question.

**Comment by roryqi** [comment]
@advancedxy @LuciferYang @rickyma @maobaolong Thanks for your comments, I have addressed all the comments.

File: community/maturity.md (hunk 1 above)
Comment (roryqi) [reply community/maturity.md]: reformatted.

File: community/maturity.md (hunk 7 above)
Comment (advancedxy) [reply community/maturity.md]: ok, then it's fine to me.

**Review by advancedxy — COMMENTED** [review]
LGTM

**Review by advancedxy — APPROVED** [review]

**Review by kaijchen — APPROVED** [review]
LGTM

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Review by rickyma — APPROVED** [review]
LGTM

File: community/maturity.md [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+
+### Independence
+
+| **ID**   | **Description**                                                                          | **Status**                                                                                                                                                                   |
+| -------- | ---------------------------------------------------------------------------------------- |------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| **IN10** | The project is independent from any corporate or organizational influence.               | **YES** The PPMC members and committers of Apache Uniffle are from several different companies, and the majority of them are NOT From the company that donated this project. |
```
Comment (rickyma) [inline community/maturity.md]: nit: From -> from

Comment (roryqi) [reply community/maturity.md]: Addressed.

**Review by maobaolong — APPROVED** [review]
+1

**Comment by roryqi** [comment]
Thanks all. Merged.

**Pull request merged**